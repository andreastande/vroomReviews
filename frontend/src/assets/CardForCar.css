.car-image {
  height: 12rem;
  border-radius: 1rem;
}

.card {
  display: flex;
  justify-content: center;
  margin: 0;
}

.card :hover {
  transform: scale(1.05);
  transition: all 0.2s ease-in-out;
}
