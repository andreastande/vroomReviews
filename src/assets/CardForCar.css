.car-image {
  width: 20rem;
  border-radius: 1rem;
}

.card :hover {
  transform: scale(1.05);
  transition: all 0.2s ease-in-out;
}

.car-name {
  margin-left: 5rem;
}