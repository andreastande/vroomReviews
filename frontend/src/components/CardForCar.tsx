import { Link } from 'react-router-dom';
import '../assets/CardForCar.css';

function CardForCar({
  brand,
  model,
  carIMG,
  showInfo,
}: {
  brand: string | null;
  model: string | null;
  carIMG: string;
  showInfo: boolean;
}) {
  return (
    <div>
      <div className="card">
        <Link to={`/project2/carpage/${brand}-${model}`}>
          <img src={carIMG} className="car-image" alt={`${brand}-${model}`} />
        </Link>
      </div>
      {showInfo ? (
        <div className="car-name">
          <div className="car-title-wrapper">
            <h1 className="car-title">{brand}</h1> 
            <p className="car-title">{model}</p>
          </div>
        </div>
      ) : null}
    </div>
  );
}
export default CardForCar;
