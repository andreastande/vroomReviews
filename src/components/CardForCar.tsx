import { Link } from 'react-router-dom';
import '../assets/CardForCar.css';

function CardForCar({
  id,
  brand,
  model,
  carIMG,
  showInfo,
}: {
  id: string;
  brand: string;
  model: string;
  carIMG: string;
  showInfo: boolean;
}) {
  return (
    <div >
      <div className="card">
      <Link to={`/carpage/${id}`}>
        <img src={carIMG} className="car-image" />
      </Link>
      </div>
      {showInfo ? (
        <div className="car-name">
          <h1>
            {brand} {model}
          </h1>
        </div>
      ) : null}
    </div>
  );
}
export default CardForCar;
