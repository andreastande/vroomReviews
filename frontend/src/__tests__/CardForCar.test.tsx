import { render } from '@testing-library/react';
import { test, expect } from 'vitest';
import userEvent from '@testing-library/user-event';
import CardForCar from '../components/CardForCar';
import { BrowserRouter } from 'react-router-dom';

test('Check path correct when image is clicked', async () => {
  const { getByRole, asFragment } = render(
    <BrowserRouter>
      <CardForCar
        brand="Ferrari"
        model="F40"
        carIMG="https://upload.wikimedia.org/wikipedia/commons/thumb/c/cb/F40_Ferrari_20090509.jpg/1200px-F40_Ferrari_20090509.jpg"
        showInfo={false}
      />
    </BrowserRouter>,
  );

  const carimg = getByRole('img');
  expect(carimg).toBeTruthy(); // The text is found
  await userEvent.click(carimg);
  expect(asFragment()).toMatchSnapshot();
  const { pathname } = location;

  expect(pathname).toBe('/project2/carpage/Ferrari-F40');
});
