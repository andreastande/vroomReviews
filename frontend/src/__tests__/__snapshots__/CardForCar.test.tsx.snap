// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Check path correct when image is clicked 1`] = `
<DocumentFragment>
  <figure
    class="card"
  >
    <a
      href="/project2/carpage/Ferrari-F40"
    >
      <img
        alt="Ferrari-F40"
        class="car-image"
        src="https://upload.wikimedia.org/wikipedia/commons/thumb/c/cb/F40_Ferrari_20090509.jpg/1200px-F40_Ferrari_20090509.jpg"
      />
    </a>
  </figure>
</DocumentFragment>
`;
